### ISSUE TITLE
SafetyNet failed [ctsProfile] 

### ISSUE THREAD

**Issue by iSynex** [kdrag0n/safetynet-fix#78]
Google released a fix tonight
As I understand it, in order to pass CTSprofile, you need certification on the TPM module
[<img width="300" height="500" src="https://i.imgur.com/VmNeOIM.png" >](Error)

**Comment by Catfriend1** [comment]
I'm  seeing cts profile failed on pixelexperience since today, without the kdragon mod installed.

**Comment by dmirubtsov** [comment]
https://telegra.ph/Pass-Safetynet-on-Rooted-Device-09-02-2

This instruction helped me -- disallow tracking for google play services in XPrivacyLua and clear data. Please check

**Comment by AndreasAZiegler** [comment]
@dmirubtsov Thanks for the hint, but the guide doesn't work for my Nokia 6.1 with LineageOS 18.1

**Comment by dmirubtsov** [comment]
@AndreasAZiegler I tested it on my Note9 with LineageOS 18.1 and it works

Please  double check the steps. Need to add play services to both lsp and xprivacylua -- easy to miss it

Also I use 'google pixel 2' fingerprint.

Maybe something is different on your device. If it don't work let's check comments from another guys

**Comment by AnyoeDove** [comment]
@dmirubtsov Thanks! It works on my Pixel 3! Not sure how or why tho haha.

_[Use tracking](https://github.com/M66B/XPrivacyLua/blob/master/README.md) (fake user agent for WebView only, Build properties, network/SIM country/operator)_

**Comment by AndreasAZiegler** [comment]
@dmirubtsov Thanks for your help. I did not set up https://github.com/Magisk-Modules-Repo/MagiskHidePropsConf properly. It is now working.

**Comment by prochanunez17** [comment]
@dmirubtsov Worked on my Pixel 4a 5G running LineageOS 18.1 thank you :)

**Comment by Catfriend1** [comment]
The above posted telegraph solution worked for me. Thosw steps were enough.  I didn't trust the attached zip download, so I went into magisk's module tab with the repository and grabbed lsposed and riru from there. Got the xprivacylua app directly from fdroid.

```
-- Open Magisk and Flash Riru Core, But do not reboot.

 -- Go back and flash Lsposed in magisk & reboot your device. 

 -- Now install XPrivacyLua apk

 -- Open Lsposed and go to modules section and choose XPrivacyLua from there.

 -- Select system framework, Settings storage and Google Play services there.(Uncheck the system apps from hide section settings for Play services to show up)

-- Reboot your device

 -- Open XPrivacyLua and search google play services, click drop down menu in left of play services and tick 'use tracking' as shown in the screenshot.

-- Now clear play services and play store data through your phone's settings. 
```

Now gpay is working again. thanks!

**Comment by arifpedia** [comment]
@dmirubtsov SafetyNet test in magisk is pass, but SafetyNet test in other app failed, because its can't call Google if we block via XPrivacyLua

![Screenshot_2021-09-03-07-52-57-463_ya a axd jnm](https://user-images.githubusercontent.com/4081293/131934301-a19aa88f-bb43-4d68-8190-77a0aca0e7ef.jpg)
![IMG_20210903_074844](https://user-images.githubusercontent.com/4081293/131934306-c321d1de-5625-4429-89e8-acba24486139.jpg)

**Comment by SBence** [comment]
I believe the bug is in that SafetyNet checker app. I've tried two others and they both reported a SafetyNet pass.

**Comment by arifpedia** [comment]
@SBence : I think so. I have deleted my comment

**Comment by zzahkaboom24** [comment]
@arifpedia yep, i tried 6 checker apps including the safetynet checker app and magisk, all 5 of them passed except the one you mentioned

**Comment by R4wd0g** [comment]
I did the @dmirubtsov procedure and got success.
Additionally, I installed this magisk module to make Google pay work: GPay-SQLite-Fix
https://github.com/stylemessiah/GPay-SQLite-Fix

**Comment by dmirubtsov** [comment]
@R4wd0g it works on my Note9 without that

Maybe it depends on device/os

**Comment by non-bin** [comment]
The XPrivacyLua method didn't work for me :(

Pixel 4a (tried stock & lineage)

**Comment by BlueBartender** [comment]
> I believe the bug is in that SafetyNet checker app. I've tried two others and they both reported a SafetyNet pass.

Agreed. The app I was using was failing because it "can't access google play services".  After trying and retrying the Lua settings, I finally tried a couple other apps, and all was well.   Working on Pixel 5.

**Comment by dhammel** [comment]
@Catfriend1 Works flawlessly on my pixel 4, safteynet now success once again. Havent tested yet though, but I believe this safteynet-fix module still needs to be installed and enabled.

Thank you

**Comment by NoXeDev** [comment]
There is youtube video. That work for me : https://www.youtube.com/watch?v=I9A3yxAi-Ww

**Comment by Catfriend1** [comment]
@dhammel  I did not install the safety net fix module and it works.

**Comment by dhammel** [comment]
@Catfriend1 Oh, ok. Then im going to later see what part of the process is actually bypassing safteynet, it looked like as part of xuaprivacylua that it was restricting tracking, figured the wrest was handled by this module, maybe i was wrong. Ill disable it later, reboot and re-run safteynet test to see my results.

This cat and mouse game is still ongoing it seems, for roughly 8 months this module always worked, hopefully theres a fix developer can come up with for this module.

**Comment by huxley101** [comment]
You are all such beautiful people helping to solve these problems. Gpay is back in business now after I followed the instructions above.  (I also had to re add my cards but it looks happy now) I have no idea if running the xprivacy etc will slow my phone down but root is absolutely essential for a developer and I don't see why google should be punishing me when there really is no risk to security.

**Comment by doclucas** [comment]
It's a bit funny that the solution/workaround posted here actually got nothing to do with the excellent @kdrag0n module and works even without it installed.
Hopefully @kdrag0n can upgrade safetynet-fix to make this XPrivacyLua solution unnecessary.

**Comment by bryce-lynch** [comment]
> Hopefully @kdrag0n can upgrade safetynet-fix to make this XPrivacyLua solution unnecessary.

[Adding model props](https://forum.xda-developers.com/t/magisk-general-support-discussion.3432382/page-2375#post-85573473) to kdrag0n's fix makes it work without any need for XPrivacyLua.

**Comment by qqUber** [comment]
@

> I reproduced the @dmirubtsov tip and got success.
> Additionally, I installed this magisk module to make Google pay work: GPay-SQLite-Fix
> https://github.com/stylemessiah/GPay-SQLite-Fix

Thx bro, it's work on mi9t with latest pixel exp ui A11

**Comment by bongochong** [comment]
Using MagiskHide Props Config to delete the ro.product.model prop was the only method that fixed this for me. Worked like a charm. I imagine that it could cause issues on certain devices, but it was much less time consuming than the method involving Riru, and it actually worked, unlike the safetynet-fix module uploaded by Displax.

**Comment by non-bin** [comment]
> > Hopefully @kdrag0n can upgrade safetynet-fix to make this XPrivacyLua solution unnecessary.
> 
> [Adding model props](https://forum.xda-developers.com/t/magisk-general-support-discussion.3432382/page-2375#post-85573473) to kdrag0n's fix makes it work without any need for XPrivacyLua.

This worked for my P4a with lineage!

**Comment by bryce-lynch** [comment]
Also: https://forum.xda-developers.com/t/magisk-general-support-discussion.3432382/post-85573141

**Comment by guoyanhuyun** [comment]
> https://telegra.ph/Pass-Safetynet-on-Rooted-Device-09-02-2
> 
> This instruction helped me -- disallow tracking for google play services in XPrivacyLua and clear data. Please check

worked on Pixel 3 Android 11

**Comment by Conan179** [comment]
> disallow tracking for google play services in XPrivacyLua and clear data. Please check

Working on note 20 ultra 5g and note 9.
I am more surprised that it works, that it is so eager to pass the check ...

**Comment by Guroff1974** [comment]
[safetynet-fix-v1.2.0-MOD.zip](https://github.com/kdrag0n/safetynet-fix/files/7110406/safetynet-fix-v1.2.0-MOD.zip)

**Comment by Abolfazl** [comment]
> [safetynet-fix-v1.2.0-MOD.zip](https://github.com/kdrag0n/safetynet-fix/files/7110406/safetynet-fix-v1.2.0-MOD.zip)

Just an FYI, this zip file comes directly from https://forum.xda-developers.com/t/magisk-general-support-discussion.3432382/post-85573473 (MD5 hashes match for the files). I just tested it on my Pixel 5 and was able to have SafetyNet pass without having to install riru, lsposed, or XPrivacyLua.

**Comment by gordboy** [comment]
The method by @bongochong a few posts back worked great for me.

1. Install USNF and MHPC modules
2. reboot and run props in a root shell
3. choose a fingerprint
4. delete the ro.product.model key
5. reboot

Of course, Google will break everything again just when we thought all was well ... :)

Samsung S5e (gts4lvwifi) LOS 18.1

**Comment by S-A-Mayer** [comment]
@gordboy You're right about the next change breaking things. I'm really pessimistic about our chances given that topjohnwu [announced last week that MagiskHide is EoL](https://topjohnwu.medium.com/state-of-magisk-2021-fe29fdaee458).

**Comment by Catfriend1** [comment]
Does v2.0.0 also fix this issue? @kdrag0n

**Comment by Abuzarar** [comment]
I used this but it is not working

**Comment by kdrag0n** [comment]
Fixed in [v2.1.0](https://kdrag0n.dev/patreon/safetynet-fix/v2.1.0?utm_source=github) without affecting UX or breaking device-specific features.

**Comment by Abuzarar** [comment]
It is not downloading give a gmail link please

**Comment by Catfriend1** [comment]
Download costs money - will take two days of consideration

**Comment by stamepicmorg** [comment]
> give a gmail link please

or you can support developer and buy preview-version with 2 bucks

**Comment by Spycoal** [comment]
> Fixed in [v2.1.0](https://kdrag0n.dev/patreon/safetynet-fix/v2.1.0?utm_source=github) without affecting UX or breaking device-specific features.

Is there an ETA when 2.1.0 leaves preview? Not everybody (including myself) can subscribe but would still be interested in the fix.
